core/domain/exp_services.py

```diff
@@ -650,34 +652,41 @@ def _create_exploration(
         param_specs=exploration.param_specs_dict,
         param_changes=exploration.param_change_dicts,
         auto_tts_enabled=exploration.auto_tts_enabled,
-        correctness_feedback_enabled=exploration.correctness_feedback_enabled
-    )
+        correctness_feedback_enabled=exploration.correctness_feedback_enabled)
+
+    exploration.version += 1
     commit_cmds_dict = [commit_cmd.to_dict() for commit_cmd in commit_cmds]
 
-    def _update_storage_models_transactional():
-        """Commits all the newly created models as an atomic transaction. If any
-        operation fails, every other operation will fail as well.
+    new_exp_stats = stats_services.get_stats_for_new_exploration(
+        exploration.id, exploration.version, exploration.states)
+
+    def _update_storage_models():
+        """Groups all storage calls into a function call so that they can be
+        performed as an atomic transaction.
         """
+        # Rights to the exploration need to be created before the actual
+        # exploration.
         rights_manager.create_new_exploration_rights(
             exploration.id, committer_id)
-        model.commit(committer_id, commit_message, commit_cmds_dict)
-        exploration.version += 1
+
+        exploration_model.commit(committer_id, commit_message, commit_cmds_dict)
         create_exploration_summary(exploration.id, committer_id)
 
-        stats_services.create_stats_model(
-            stats_services.get_stats_for_new_exploration(
-                exploration.id, exploration.version, exploration.states))
+        stats_services.create_stats_model(new_exp_stats)
+
         stats_services.create_exp_issues_for_new_exploration(
             exploration.id, exploration.version)
 
-    transaction_services.run_in_transaction(
-        _update_storage_models_transactional)
+    transaction_services.run_in_transaction(_update_storage_models)
 
     if feconf.ENABLE_ML_CLASSIFIERS:
-        state_names_to_train = [
-            state_name for state_name, state in exploration.states.items()
-            if state.can_undergo_classification()
-        ]
+        # Find out all states that need a classifier to be trained.
```

Comment:
Yeah I reverted them because I couldn't get the tests to pass due to these refactorings. I decided to keep them out to make sure this PR goes out ASAP.